lms/djangoapps/instructor/views/api.py

```diff
@@ -970,6 +971,76 @@ def extract_user_info(user):
     return JsonResponse(response_payload)
 
 
+@view_auth_classes()
+@method_decorator(transaction.non_atomic_requests, name='dispatch')
+class ProblemResponseReportInitiate(DeveloperErrorViewMixin, APIView):
+    """
+    Initiate generation of a CSV file containing all student answers
+    to a given problem.
+
+    **Example requests**
+
+        POST /courses/course-v1:edX+DemoX+Demo_Course/instructor/api/v0/generate_report/problem_responses {
+            "problem_location": "{usage_key1},{usage_key2},{usage_key3}""
+        }
+        POST /courses/course-v1:edX+DemoX+Demo_Course/instructor/api/v0/generate_report/problem_responses {
+            "problem_location": "{usage_key}",
+            "problem_types_filter": "problem"
+        }
+
+    **POST Parameters**
+
+    A POST request can include the following parameters:
+
+    * problem_location: A comma-separated list of usage keys for the blocks
+      to include in the report. If the location is a block that contains
+      other blocks, (such as the course, section, subsection, or unit blocks)
+      then all blocks under that block will be included in the report.
+    * problem_types_filter: Optional. A comma-separated list of block types
+      to include in the repot. If set, only blocks of the specified types will
+      be included in the report.
+
+    To get data on all the poll and survey blocks in a course, you could
+    POST the usage key of the course for `problem_location`, and
+    "poll, survey" as the value for `problem_types_filter`.
+
+
+    **Example Response:**
+    If initiation is successful (or generation task is already running):
+    ```json
+    {
+        "status": "The problem responses report is being created. ...",
+        "task_id": "4e49522f-31d9-431a-9cff-dd2a2bf4c85a"
+    }
+    ```
+
+    Responds with BadRequest if any of the provided problem locations are faulty.
+    """
+    @apidocs.schema(parameters=[
+        apidocs.string_parameter(
+            'course_id',
+            apidocs.ParameterLocation.PATH,
+            description="ID of the course for which report is to be generate.",
+        ),
+        apidocs.string_parameter(
+            'problem_location',
+            apidocs.ParameterLocation.BODY,
+            description="One or more usage keys for blocks whose report is needed, separated by commas.",
+        ),
+        apidocs.string_parameter(
+            'problem_types_filter',
+            apidocs.ParameterLocation.BODY,
+            description="One or more block types separated by commas. "
+                        "The generated report will be limited to the specified block types",
+        ),
+    ])
+    def post(self, request, course_id):
+        """
+        Initiates asynchronous problem response report generation.
+        """
+        return get_problem_responses(request=request, course_id=course_id)
```

Comment:
I'm just directly using the existing view method since it already does what we want with the exception of the auth methods supported.